js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -866,7 +878,9 @@ export const validatingModule = {
                         }
 
                         if(shouldResetValidationState) {
-                            this.getController('validating')._validationState = [];
+                            const validatingController = this.getController('validating');
+                            validatingController._validationState = [];
+                            validatingController._validationStateCache = {};
```

Comment:
```suggestion
                            validatingController.resetValidationState();
```
I would put these two lines in a separate method and use it instead.